src/main/java/picoded/core/conv/DateConv.java

```diff
@@ -1,237 +1,92 @@
-		String separator) {
-		if (inDate == null || StringUtils.isEmpty(inDate)) {
-			return null;
-		}
-		
-		String newDate = changeISODateFormat(inDate, currentDateFormat, ISODateFormat.YYYYMMDD,
-			separator);
-		String[] newDateSplit = null;
-		if (newDate != null) {
-			newDateSplit = newDate.split(separator);
-		}
-		if (newDateSplit == null) {
-			return null;
-		}
-		
-		Calendar cal = Calendar.getInstance();
-		cal.set(Integer.parseInt(newDateSplit[0]), Integer.parseInt(newDateSplit[1]) - 1,
-			Integer.parseInt(newDateSplit[2]));
-		return String.valueOf(cal.getTimeInMillis());
-		
-	}
+public class DateConv extends DateUtils {
 
-	/**
-	 * Date to Seconds conversion function
-	 * @param inDate
-	 * @param currentDateFormat
-	 * @param separator
-	 * @return
-	 */
-	public static String toSecondsFormat(String inDate, ISODateFormat currentDateFormat,
-											  String separator) {
-		// Code reusing : toMillisecondsFormat
-		String dateInMiliSeconds = toMillisecondsFormat(inDate, currentDateFormat, separator);
-		if (dateInMiliSeconds == null) {
-			return null;
-		}
-		long longDate = Long.parseLong(dateInMiliSeconds);
+    /**
+     * Invalid constructor (throws exception)
+     **/
+    protected DateConv() {
+        throw new IllegalAccessError(ExceptionMessage.staticClassConstructor);
+    }
 
-		return String.valueOf(longDate/1000L);
-	}
+    /**
+     * Convert a Date from String to java.util.Date Class
+     * Given a Date as a String and its pattern
+     * Then, this function will build the Java Date object out of that  DateString and pattern
+     *
+     * @param dateStr
+     * @param pattern
+     * @return
+     */
+    public static Date fromDateString(final String dateStr, final String pattern) {
+        try {
+            return DateUtils.parseDate(dateStr, new String[]{pattern});
+        } catch (ParseException e) {
+            throw new RuntimeException(e);
+        }
+    }
 
+    /**
+     * Convert a java.util.Date Object to String
+     * Specify the Date format and provide java.util.Date Object in the parameter
+     * This function would constructs Date String according to the format given
+     * ```
+     * // Conversion to a date string, from milliseconds timestamp
+     * DateConv.toDateString( new Date(timestamp), "YYYY-MM-DD" );
+     * ```
+     *
+     * @param date
+     * @param outputPattern
+     * @return
+     */
+    public static String toDateString(final Date date, final String outputPattern) {
+        SimpleDateFormat formatter = new SimpleDateFormat(outputPattern);
+        return formatter.format(date);
+    }
 
-	
-	//Util functions
+    /**
+     * Convert Unix timestamp to Date in String according to the output pattern
+     *
+     * @param timestamp
+     * @param outputPattern
+     * @return
+     */
+    public static String fromUnixTimestamp(final long timestamp, final String outputPattern) {
```

Comment:
`DateConv.fromUnixTimestamp` implies it outputs as a `Date` object, not a string. You can simply just rename this to `DateConv.toDateString`, and let type overloading handle the difference variable type properly.